compose/compiler/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/lower/ComposableFunctionBodyTransformer.kt

```diff
@@ -3145,6 +3145,7 @@ class ComposableFunctionBodyTransformer(
         var needsWrappingGroup = false
         var someResultsHaveCalls = false
         var hasElseBranch = false
+        var multipleResultsHaveCalls = false
```

Comment:
some of the code here might be easier to reason about if we instead kept a count of results with calls `var resultsWithCalls: Int = 0`. The conditions later down the page would simplify a lot where we could say `if (resultsWithCalls == 1) ...` or `if (resultsWithCalls > 1) ...`